src/main/kotlin/lotto/domain/PurchaseSession.kt

```diff
@@ -0,0 +1,57 @@
+package lotto.domain
+
+import lotto.exceptions.LottoException.InvalidAmountException
+import lotto.exceptions.LottoException.InvalidAutomaticTicketsException
+import lotto.exceptions.LottoException.InvalidManualTicketsException
+import lotto.exceptions.LottoException.InvalidManualTicketsNumberException
+
+data class PurchaseSession(
+    val amount: Int = 0,
+    val allTicketsNumber: Int = 0,
+    val manualTicketsNumber: Int = 0,
+    val automaticTickets: List<Lotto> = emptyList(),
+    val manualTickets: List<Lotto> = emptyList(),
+    val winningCombination: WinningCombination? = null,
+    val ticketsRank: List<Rank> = emptyList(),
+    val returnRate: Double = 0.0,
+) {
+    val automaticTicketsNumber: Int
+        get() = allTicketsNumber - manualTicketsNumber
+    val allTickets: List<Lotto>
+        get() = manualTickets + automaticTickets
+
+    fun updateAmount(amount: Int): PurchaseSession {
+        val condition = amount >= LottoMachine.PRICE_OF_TICKET && amount % LottoMachine.PRICE_OF_TICKET == 0
+        require(condition && amount <= LottoMachine.MAX_AMOUNT_ACCEPTED) {
```

Comment:
Only a part of the whole condition was put in the variable, while the last one still exists in `require` block. Also, since there are multiple conditions here, it's kinda hard to understand it right away. Just by dividing them into smaller parts and giving them meaningful name, we'll be able to make it more readable : 
```suggestion
        val isAboveMinimum = amount >= LottoMachine.PRICE_OF_TICKET
        val isDivisible = amount % LottoMachine.PRICE_OF_TICKET == 0
        val isBelowMaximum = amount <= LottoMachine.MAX_AMOUNT_ACCEPTED

        require(isAboveMinimum && isDivisible && isBelowMaximum) {
            throw InvalidAmountException(amount)
        }
```

Reply:
Yes! Totally, that make it completely more readable. Thanks!